src/qt/walletmodel.cpp

```diff
@@ -756,3 +779,296 @@ bool WalletModel::getDefaultWalletRbf() const
 {
     return fWalletRbf;
 }
+
+bool WalletModel::nameAvailable(const QString &name)
+{
+    JSONRPCRequest jsonRequest;
+    UniValue params (UniValue::VOBJ);
+    UniValue res, array, isExpired;
+
+    const std::string strName = name.toStdString();
+    params.push_back (Pair("name", strName));
+    jsonRequest.strMethod = "name_show";
+    jsonRequest.params = params;
+    jsonRequest.fHelp = false;
+
+    try {
+        res = tableRPC.execute(jsonRequest);
```

Comment:
In general, I think that calling RPCs as you do is a nice way to decouple the code from the core name logic.  But in this case, could you simply use `pcoinsTip->GetName`?  Especially since you use that above somewhere -- I think you should either use it in both places, or call `name_show` in both places -- unless there are differences between here and above that I overlooked.

Reply:
I changed the `GetName` call in `NameTableModel` to use the RPC like the rest of the GUI.

Reply:
Thanks, looks good.